docs/usage/shoot_high_availability_best_practices.md

```diff
@@ -0,0 +1,431 @@
+
+You will have to create specific `HorizontalPodAutoscaler` resources for your scale target and can [tweak the general HPA knobs](https://gardener.cloud/docs/gardener/api-reference/core/#horizontalpodautoscalerconfig) for Gardener-managed clusters like this:
+
+``` yaml
+apiVersion: core.gardener.cloud/v1beta1
+kind: Shoot
+spec:
+  kubernetes:
+    kubeControllerManager:
+      horizontalPodAutoscaler:
+        syncPeriod: 30s
+        tolerance: 0.1
+        downscaleStabilization: 5m0s
+        initialReadinessDelay: 30s
+        cpuInitializationPeriod: 5m0s
+```
+
+## Resources (Vertical Scaling)
+
+While it is important to set a sufficient number of replicas, it is also important to give the pods sufficient resources (CPU and memory). This is especially true when you think about HA. When a zone goes down, you might need to get up replacement pods, if you don't have them running already to take over the load from the impacted zone. Likewise, e.g. with active-active software components, you can expect the remaining pods to receive more load. If you cannot scale them out horizontally to serve the load, you will probably need to scale them out vertically. This is done by the [vertical pod autoscaler](https://github.com/kubernetes/autoscaler/tree/master/vertical-pod-autoscaler) or VPA for short (not built-in; part of the autoscaler repository).
+
+A few caveats though:
+
+- You cannot use HPA and VPA on the same metrics as they would influence each other, which would lead to pod trashing (more replicas require fewer resources; fewer resources require more replicas)
+- Scaling horizontally doesn't cause downtimes (at least not when out-scaling and only one replica is affected when in-scaling), but scaling vertically does (if the pod runs OOM anyway, but also when new recommendations are applied, resource requests for existing pods may be changed, which causes the pods to be rescheduled). Although the discussion is going on for a very long time now, that is still not supported in-place yet (see [KEP 1287](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/1287-in-place-update-pod-resources/README.md), [implementation in Kubernetes](https://github.com/kubernetes/kubernetes/pull/102884), [implementation in VPA](https://github.com/kubernetes/autoscaler/issues/4016)).
+
+VPA is a useful tool and Gardener-managed clusters deploy a VPA for you (HPA is supported anyway as it's built into the kube-controller-manager). You will have to create specific `VerticalPodAutoscaler` resources for your scale target and can [tweak the general VPA knobs](https://gardener.cloud/docs/gardener/api-reference/core/#verticalpodautoscaler) for Gardener-managed clusters like this:
```

Comment:
Nit: VPA is enabled by default, Gardener does not deploy VPA unconditionally.

```suggestion
VPA is a useful tool and Gardener-managed clusters deploy a VPA by default for you (HPA is supported anyway as it's built into the kube-controller-manager). You will have to create specific `VerticalPodAutoscaler` resources for your scale target and can [tweak the general VPA knobs](https://gardener.cloud/docs/gardener/api-reference/core/#verticalpodautoscaler) for Gardener-managed clusters like this:
```

Reply:
OK, added.